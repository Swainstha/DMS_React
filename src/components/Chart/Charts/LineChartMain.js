import React from 'react';
import {CartesianGrid, Legend, Line, LineChart, ResponsiveContainer, Tooltip, XAxis, YAxis} from 'recharts';

const LineChartWithXAxisPading = (data) => (
  <ResponsiveContainer width="100%" height={data.height} data-test="responsive-component">
    <LineChart data-test="linechart-component" data={data.data} margin={{top: 10, right: 0, left: -15, bottom: 0}}>
      <XAxis dataKey="month" padding={{left: 30, right: 30}}/>
      <YAxis/>
      <CartesianGrid strokeDasharray="3 3"/>
      <Tooltip/>
      <Legend/>
      <Line type="monotone" dataKey="deliveries" stroke="#3367d6" activeDot={{r: 8}}/>
      {/* <Line type="monotone" dataKey="Drone" stroke="#ffc658"/> */}
    </LineChart>
  </ResponsiveContainer>
)

export default LineChartWithXAxisPading