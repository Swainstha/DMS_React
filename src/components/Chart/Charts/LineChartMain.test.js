import React from 'react';
import LineChartMain from './LineChartMain';
import {
    configure,
    shallow,
} from 'enzyme';
import EnzymeAdapter from 'enzyme-adapter-react-16';
// JestHook.mock('expo-font');
configure({
    adapter: new EnzymeAdapter
})
const setup = (props = {}) => {
    return shallow( < LineChartMain {...props}  />
    )
}

const findByTestAttr=(wrapper, val)=>{
    return wrapper.find(`[data-test='${val}']`)
}


it("LineChartMain",()=>{
    const wrapper= setup({data:[2,3,4], height:300});
    expect(wrapper).toMatchSnapshot();
})

it("LineChartMain props",()=>{
    const wrapper= setup({"data":[2,3,4], height:300});
    // let child0= findByTestAttr(wrapper, 'child0')
    let child1= findByTestAttr(wrapper, 'linechart-component')
    expect(child1).toHaveLength(1)
    expect(child1.props("data").data).toEqual([2,3,4])
})

it("LineChartMain props",()=>{
    const wrapper= setup({"data":[2,3,7], height:300});
    // let child0= findByTestAttr(wrapper, 'child0')
    let child1= findByTestAttr(wrapper, 'linechart-component')
    let child2= findByTestAttr(wrapper, 'responsive-component')
    expect(child1.props("data").data).toEqual([2,3,7])
    expect(child1).toHaveLength(1)
    expect(child2.props("data").height).toEqual(300)
})