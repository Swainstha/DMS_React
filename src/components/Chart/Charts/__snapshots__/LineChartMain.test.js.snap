// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LineChartMain 1`] = `ShallowWrapper {}`;
